lightning-invoice/src/payment.rs

```diff
@@ -1459,11 +1922,20 @@ mod tests {
 
 	impl Score for TestScorer {
 		fn channel_penalty_msat(
-			&self, _short_channel_id: u64, _source: &NodeId, _target: &NodeId, _usage: ChannelUsage
-		) -> u64 { 0 }
+			&self, _short_channel_id: u64, _source: &NodeId, _target: &NodeId, usage: ChannelUsage
+		) -> u64 {
+			match self.scorer_expectations.borrow_mut().pop_front() {
+				Some(expectation) => {
+					assert_eq!(expectation.amount_msat, usage.amount_msat);
+					assert_eq!(expectation.inflight_htlc_msat, usage.inflight_htlc_msat);
+				},
+				None => {},
```

Comment:
I think we want an `Option` around `scorer_expectations` with `None` representing no expectations set (i.e., no checks). Otherwise, you can exhaust all expectations and not check these assertions if the scorer is still called.